main.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+    "errors"
+    "fmt"
+)
+
+type Node struct {
```

Comment:
```Node``` type is exported here. If you do not want to export, use lowercase ```node```

Reply:
Also, there is a convention to use  a comment in form of ```// Node represents or is ``` before any exported type.